### PR TITLE
Use native apt wildcards and apt pinning

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
Instead of using `last-version`, just using the natively supported `apt` wildcards (v1.8.x becomes v1.8.* and you can pass that version
directly to `apt`).

### CODE REVIEW

File: docker/util.sh [hunk 1]
```diff
@@ -5,20 +5,42 @@
...
-apt_ver()
+# This function installs packages with specific versions and also pin them
+# (with prio 990) to those versions by appending pinning information to
+# /etc/apt/preferences.d/cachalot, so they are not upgraded accidentally.
+#
+# All packages should be specified with version: pkg=version.
+#
+# Version can have wildcards, see apt-get/apt_preferences documentation for
+# details
+apt_pin_install()
```
Comment (nemanja-boric-sociomantic) [inline docker/util.sh:16]: This doesn't deserve separate commit?

Comment (leandro-lucarella-sociomantic) [reply docker/util.sh:16]: Yeah, maybe. Is true that pinning and apt wildcards are not the same, but OTOH if I separate it, I will have to put a normal `apt install` call in `docker/dlang` just to change it in the next commit.

Comment (leandro-lucarella-sociomantic) [reply docker/util.sh:16]: OK, done.

**Comment by nemanja-boric-sociomantic** [comment]
Conflicting files ;-)

**Comment by leandro-lucarella-sociomantic** [comment]
And rebased.

File: docker/util.sh [hunk 2]
```diff
@@ -5,6 +5,42 @@
...
+{
+	( { set +x -e; } 2>/dev/null # dissable verboseness (and silently)
```
Comment (nemanja-boric-sociomantic) [inline docker/util.sh]: `(and silently)`? also typo `dissable`

File: docker/util.sh [hunk 3]
```diff
@@ -5,6 +5,42 @@
...
+	for pkg_ver in "$@"
+	do
+		pkg="$(echo "$pkg_ver" | cut -d= -f1)"
+		ver="$(echo "$pkg_ver" | cut -d= -f2-)"
+
+		# If we still have a version, print the pinning spec to Prio
+		# 990 means don't install newer versions, but don't downgrade
+		# if installed is newer.
+		# https://manpages.debian.org/stretch/apt/apt_preferences.5.en.html#How_APT_Interprets_Priorities
+		cat <<EOT >> /etc/apt/preferences.d/cachalot
```
Comment (nemanja-boric-sociomantic) [inline docker/util.sh:30]: Do we want to do `ftruncate -s0 /etc/apt/preferences.d/cachalot` just in case first?

**Comment by nemanja-boric-sociomantic** [comment]
In general LGTM!

File: docker/util.sh (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [reply docker/util.sh]: "Disables verboseness silently". You need a trick not to get a `+set +x` printed if `set -x` was set before :)

Comment (nemanja-boric-sociomantic) [reply docker/util.sh]: Ah, I got it now.

File: docker/util.sh (hunk 3 above)
Comment (leandro-lucarella-sociomantic) [reply docker/util.sh:30]: The answer is no. Otherwise if we call this command twice it will undo what the previous command did. I think is safe to assume that upstream ubuntu won't ship a `/etc/apt/preferences.d/cachalot`, but we could do the tructation in the base image if we are really concerned about this.

File: docker/util.sh (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [reply docker/util.sh]: Added commit to fix the typos (there was the same type in other parts of this file)

File: docker/util.sh (hunk 3 above)
Comment (nemanja-boric-sociomantic) [reply docker/util.sh:30]: You're right, we're calling this function several times per image generation.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

File: docker/util.sh (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [reply docker/util.sh]: Maybe I could remove the **and** (`# disable verboseness (silently)`) as part of the typo fixing commit. Do you think is more clear that way?

Comment (nemanja-boric-sociomantic) [reply docker/util.sh]: I think it's clearer that way, but I don't think it makes much difference.

Comment (leandro-lucarella-sociomantic) [reply docker/util.sh]: Done.

File: docker/util.sh (hunk 3 above)
Comment (leandro-lucarella-sociomantic) [reply docker/util.sh:30]: Not yet in reality, but we might in the future :)
